src/uhttp.c

```diff
@@ -320,7 +321,36 @@ static int write_text_line(HTTP_CLIENT_HANDLE_DATA* http_data, const char* text_
         result = 0;
         if (http_data->trace_on)
         {
-            LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+            char* authStart = strstr(text_line, "Authorization:");
+            char* proxyAuthStart = strstr(text_line, "Proxy-Authorization");
+            if (authStart == NULL && proxyAuthStart == NULL)
+            {
+                LOG(AZ_LOG_TRACE, LOG_LINE, "%s", text_line);
+            }
+            else if (authStart != NULL && proxyAuthStart == NULL)
+            {
+                char* authEol = strstr(authStart, "\x0D\x0A");
+                if (authEol != NULL)
+                {                                                                 
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s ************** %.*s", (int)(authStart  - text_line), text_line, (int)(strlen(text_line) - (authEol - text_line)), authEol);
+                }
+            }
+            else if (proxyAuthStart != NULL && authStart == NULL)
+            {
+                char* proxyAuthEol = strstr(proxyAuthStart, "\x0D\x0A");
+                if (proxyAuthEol != NULL)
+                {
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s ************** %.*s", (int)(proxyAuthStart  - text_line), text_line, (int)(strlen(text_line) - (proxyAuthEol - text_line)), proxyAuthEol);
+                }
+            }
+            else if (proxyAuthStart != NULL && authStart != NULL)
+            {
+                char* proxyAuthEol = strstr(proxyAuthStart, "\x0D\x0A");
+                if (proxyAuthEol != NULL)
+                {
+                    LOG(AZ_LOG_TRACE, LOG_LINE, "%.*s **************\n************** %.*s", (int)(authStart  - text_line), text_line, (int)(strlen(text_line) - (proxyAuthEol - text_line)), proxyAuthEol);
```

Comment:
???
There is no position order in the header of the fields, so authStart  > proxyAuthStart

Reply:
need to handle each span separately and figure out which one comes first.

Reply:
Fixed this. Thanks!